tests/test_project.py

```diff
@@ -170,6 +174,132 @@ def helper_test_list_ccas(project):
+        assert False
+    except SherlockAddCCAError as e:
+        assert str(e) == "Add CCA error: CCA properties are invalid for CCA 0."
+
+    try:
+        project.add_cca(
+            "Test",
+            [
+                {
+                    "description": "Second CCA",
+                    "default_solder_type": "SAC305",
+                    "default_stencil_thickness": 10,
+                    "default_stencil_thickness_units": "mm",
+                    "default_part_temp_rise": 20,
+                    "default_part_temp_rise_units": "C",
+                    "guess_part_properties_enabled": False,
+                },
+            ],
+        )
+        assert False
+    except SherlockAddCCAError as e:
+        assert str(e) == "Add CCA error: CCA name is missing for CCA 0."
+
+    try:
+        project.add_cca(
+            "Test",
+            [
+                {
+                    "cca_name": "",
+                    "description": "Second CCA",
+                    "default_solder_type": "SAC305",
+                    "default_stencil_thickness": 10,
+                    "default_stencil_thickness_units": "mm",
+                    "default_part_temp_rise": 20,
+                    "default_part_temp_rise_units": "C",
+                    "guess_part_properties_enabled": False,
+                },
+            ],
+        )
+        assert False
+    except SherlockAddCCAError as e:
+        assert str(e) == "Add CCA error: CCA name is invalid for CCA 0."
+
+    if not project._is_connection_up():
+        return
+
+    try:
+        project.add_cca(
+            "Test",
+            [
+                {
+                    "cca_name": "Name",
+                    "description": "Second CCA",
+                    "default_solder_type": "SAC305",
+                    "default_stencil_thickness": 10,
+                    "default_stencil_thickness_units": "INVALID",
+                    "default_part_temp_rise": 20,
+                    "default_part_temp_rise_units": "C",
+                    "guess_part_properties_enabled": False,
+                },
+            ],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockAddCCAError as e:
+        assert type(e) == SherlockAddCCAError
+
+    cca_name = "Test Card " + str(uuid.uuid4())
+    try:
+        result = project.add_cca(
+            "Tutorial Project",
+            [
+                {
+                    "cca_name": cca_name,
+                    "description": "Second CCA",
+                    "default_solder_type": "SAC305",
+                    "default_stencil_thickness": 10,
+                    "default_stencil_thickness_units": "mm",
+                    "default_part_temp_rise": 20,
+                    "default_part_temp_rise_units": "C",
+                    "guess_part_properties_enabled": False,
+                },
+            ],
+        )
+        assert result == 0
+        time.sleep(5)
```

Comment:
Can this duration be shortened?
It is desirable to minimize the delays so that the total run time is as short as possible.